kyuubi-server/src/main/scala/org/apache/kyuubi/engine/YarnApplicationOperation.scala

```diff
@@ -95,74 +95,139 @@ class YarnApplicationOperation extends ApplicationOperation with Logging {
     }
   }
 
+  private def withNewYarnClient[T](
+      proxyUser: Option[String],
+      sessionConf: Option[KyuubiConf])(action: YarnClient => T): T = {
+    (sessionConf) match {
+      case Some(sessionConf) =>
+        (proxyUser) match {
+          case Some(user) =>
+            Utils.doAs(user) { () =>
+              var yarnClient: YarnClient = null
+              try {
+                val yarnConf = KyuubiHadoopUtils.newYarnConfiguration(sessionConf)
```

Comment:
Does this use `kyuubi.engineEnv.HADOOP_CONF_DIR/YARN_CONF_DIR` in sessionConf to initialize `yarnConf`?